src/shared/extension-release.c

```diff
@@ -119,14 +121,20 @@ int extension_release_validate(
         return 1;
 }
 
-int parse_env_extension_hierarchies(char ***ret_hierarchies) {
+int parse_env_extension_hierarchies(char ***ret_hierarchies, const char *env_hierarchy) {
         _cleanup_free_ char **l = NULL;
         int r;
 
-        r = getenv_path_list("SYSTEMD_SYSEXT_HIERARCHIES", &l);
+        assert(env_hierarchy);
+
+        r = getenv_path_list(env_hierarchy, &l);
         if (r == -ENXIO) {
-                /* Default when unset */
-                l = strv_new("/usr", "/opt");
+                if (streq(env_hierarchy, "SYSTEMD_SYSCFG_HIERARCHIES"))
+                        /* Default for syscfg when unset */
+                        l = strv_new("/etc");
+                else
+                        /* Default for sysext when unset */
+                        l = strv_new("/usr", "/opt");
```

Comment:
maybe safer:

```c
if (streq(env_hierarchy, "SYSTEMD_SYSCFG_HIERARCHIES"))
        …
else if (streq(env_hiearchy, "SYSTEMD_SYSEXT_HIERARCHIES"))
        …
else
        return -ENXIO;
```

i.e. return an error if we don't know the right default for the request env var